logstash-core/src/main/java/org/logstash/FieldReference.java

```diff
@@ -139,8 +139,11 @@ private static int calculateHash(final String key, final String[] path, final in
     }
 
     private static FieldReference parseToCache(final CharSequence reference) {
-        final FieldReference result = parse(reference);
-        CACHE.put(reference, result);
+        FieldReference result = parse(reference);
+        if (CACHE.size() < 10_000) {
```

Comment:
Can add a test for this behavior?

Reply:
@andrewvc Just run up 10k `FieldReference`s and check that `==` stops holding for `FieldRefernce` no. 10001?

Reply:
@andrewvc actually come to think about it ... this is kinda dirty to add a test for since the cache is `static` => The exact number of field references to put in here until the cache shuts down depends on the test order.
Plus (this isn't so relevant but still ...), this would actually make all tests that run after such a test slower since they're not caching anymore :P